src/app/ui/TaskBoard.tsx

```diff
@@ -59,28 +60,63 @@ export const TaskBoard = ({
   return (
     <AppMargin size={SizeofAppMargin.LARGE} py="18.5px">
       <Stack
-        direction="row"
         columnGap={2}
         sx={{
           overflowX: 'auto',
+          flexDirection: view === View.BOARD_VIEW ? 'row' : 'column',
         }}
       >
         {workflowStates.map((list, index) => {
-          return (
-            <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
-              <TaskColumn key={list.id} columnName={list.name} taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}>
-                {filterTaskWithWorkflowStateId(list.id).map((task, index) => {
-                  return (
-                    <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={task.id || ''}>
-                      <Box onClick={() => router.push(`/detail/${task.id}/iu?token=${token}`)} key={task.id} m="6px 0px">
-                        <TaskCard task={task} key={task.id} />
-                      </Box>
-                    </DragDropHandler>
-                  )
-                })}
-              </TaskColumn>
-            </DragDropHandler>
-          )
+          if (view === View.BOARD_VIEW) {
+            return (
+              <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
+                <TaskColumn
+                  key={list.id}
+                  columnName={list.name}
+                  taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}
+                >
+                  {filterTaskWithWorkflowStateId(list.id).map((task, index) => {
+                    return (
+                      <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={task.id || ''}>
+                        <Box onClick={() => router.push(`/detail/${task.id}/iu?token=${token}`)} key={task.id} m="6px 0px">
+                          <TaskCard task={task} key={task.id} />
+                        </Box>
+                      </DragDropHandler>
+                    )
+                  })}
+                </TaskColumn>
+              </DragDropHandler>
+            )
+          }
+          if (view === View.LIST_VIEW) {
+            return (
+              <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
+                <TaskRow
+                  key={list.id}
```

Comment:
This part is a bit hard to understand - also I think this is a reusable component across client and IU?

Please move this to separate components

Reply:
This is not a reusable part. We can't make use of this in the client route. For ex: client route doesn't have DnD. Also, the components like `ListViewTaskCard` and `TaskRow` needs to be passed with different props to be used in the client route. 

I think abstracting this part into another component doesn't make this any better. It will increase prop drilling for no specific reason. For instance, `key`, `accept`, `index`, `id`, `onDropItem`, `onClick`, `task`, `columnName`, `taskCount`, `showConfigurableIcons`, `updateTask`, and `handleClick` should be drilled from the parent component. Here, I am just implementing the list view and board view which is part of the `TaskBoard`.